frontend/src/views/shared/Widgets/WidgetBarChart.jsx

```diff
@@ -25,13 +34,14 @@ export const WidgetBarChart = ({
   geojson,
   type,
 }) => {
+  const t = useT();
   const { rootWidgetProps, loaded, data, noData } = useWidget(
     { slug, geojson },
     { type, analysisQuery, analysisBody },
   );
-  const { unit, bar_color } = useMemo(() => JSON.parse(legend), [legend]);
+  const { unit: singleUnit, units, bar_color } = useMemo(() => JSON.parse(legend), [legend]);
```

Comment:
Since this JSON is created by the admins, don't you think we should have a try/catch?
```suggestion
  const { unit: singleUnit, units, bar_color } = useMemo(() => {
    try {
      return JSON.parse(legend);
    } catch (e) {
      return {};
    }
  }, [legend]);
```